server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -612,7 +616,9 @@ static <T> ActionListener<T> wrapListenerForErrorHandling(
                 taskId
             );
             // Keep this logic aligned with that of SUPPRESSED_ERROR_LOGGER in RestResponse
```

Comment:
I am no longer sure how to read this comment. With this change, the two places where the logic should be aligned now differ in logic?